pkg/apis/core/v1alpha1/conversions.go

```diff
@@ -38,7 +39,24 @@ import (
 	"k8s.io/klog"
 )
 
+func init() {
+	localSchemeBuilder.Register(addConversionFuncs)
+}
+
 func addConversionFuncs(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.WithKind("Shoot"),
+		func(label, value string) (string, string, error) {
+			switch label {
+			case "metadata.name", "metadata.namespace", garden.ShootSeedName:
```

Comment:
Could you add field selector for `spec.cloudProfileName=foo`, too?

Reply:
For the old API `spec.cloud.profile=foo` might be used if you want to provide the same feature there, too

Reply:
No, it's not a requirement. Different API versions can have different field selectors.

Reply:
Added `spec.cloudProfileName=foo`.